EIPS/eip-4750.md

```diff
@@ -133,6 +133,171 @@ Alternative logic for executing `RETF` in the top frame could be to exceptionall
 
 We have decided in favor of always having at least one item in the return stack, because it allows to avoid having a special case for empty stack in the interpreter loop stack underflow check. We keep the stack underflow rule general by having `caller_stack_height = 0` in the top frame.
 
+## Reference Implementation
+
+```python
+MAGIC = b'\xEF\x00'
+VERSION = 0x01
+S_TERMINATOR = 0x00
+S_CODE = 0x01
+S_DATA = 0x02
+S_TYPE = 0x03
+
+# The ranges below are as specified in the Yellow Paper.
+# Note: range(s, e) excludes e, hence the +1
+valid_opcodes = [
+    *range(0x00, 0x0b + 1),
+    *range(0x10, 0x1d + 1),
+    0x20,
+    *range(0x30, 0x3f + 1),
+    *range(0x40, 0x48 + 1),
+    *range(0x50, 0x5d + 1),
+    *range(0x60, 0x6f + 1),
+    *range(0x70, 0x7f + 1),
+    *range(0x80, 0x8f + 1),
+    *range(0x90, 0x9f + 1),
+    *range(0xa0, 0xa4 + 1),
+    # Note: 0xfe is considered assigned.
+    *range(0xf0, 0xf5 + 1), *range(0xfa, 0xff + 1),
+]
+
+# STOP, RETURN, RETF, REVERT, INVALID, SELFDESTRUCT
+terminating_opcodes = [ 0x00, 0xf3, 0xfc, 0xfd, 0xfe, 0xff ]
+
+immediate_sizes = 256 * [0]
+immediate_sizes[0x5c] = 2  # RJUMP
+immediate_sizes[0x5d] = 2  # RJUMPI
+immediate_sizes[0xfb] = 2  # CALLF
+for opcode in range(0x60, 0x7f + 1):  # PUSH1..PUSH32
+    immediate_sizes[opcode] = opcode - 0x60 + 1
+
+class ValidationException(Exception):
+    pass
+
+# Validate EOF code.
+# Raises ValidationException on invalid code
+def validate_eof(code: bytes):
+    # Check version
+    if len(code) < 3 or code[2] != VERSION:
```

Comment:
```suggestion
    assert code[2] == VERSION, "Version must match 0x01"
```

Reply:
Consider simplifying assert

Reply:
See https://github.com/ethereum/EIPs/pull/5663#discussion_r976771485 for reasoning to use `raise`